spec/unit/matrix-client.spec.ts

```diff
@@ -52,6 +58,29 @@ jest.mock("../../src/webrtc/call", () => ({
     supportsMatrixCall: jest.fn(() => false),
 }));
 
+enum AnsiColorCode {
+    Red = 31,
+    Green = 32,
+    Yellow = 33,
+}
+// Add color to text in the terminal output
+function decorateStringWithAnsiColor(inputString: string, decorationColor: AnsiColorCode): string {
+    return `\x1b[${decorationColor}m${inputString}\x1b[0m`;
+}
+
+// Utility function to ease the transition from our QueryDict type to a string record
+// which we can use to stringify with URLSearchParams
+function convertQueryDictToStringRecord(queryDict?: QueryDict): Record<string, string> {
+    if (!queryDict) {
+        return {};
+    }
+
+    return Object.entries(queryDict).reduce((resultant, [key, value]) => {
+        resultant[key] = String(value);
+        return resultant;
+    }, {} as Record<string, string>);
+}
```

Comment:
We should be using a `Map` instead of a `Record` - please ping internally on this if you have questions.

Reply:
This doesn't seem compatible with `URLSearchParams` out of the box and I haven't bogged myself into the details on the the type compatibility here. It might be best for you to just push up a commit for me to learn from here.

What benefit are we trying to gain with `Map` as well?

Reply:
I'll have to look into this one more another time.

Reply:
What benefit are we gaining from using a `Map`? It looks like we just convert it back to an array to use it anyway

Better semantics?

Reply:
Just that it fits the semantics of what we are doing.